### PR TITLE
Chat feature

### PR DESCRIPTION
**Pull request by sp2mbn**

### CODE REVIEW

File: game/consumers.py [hunk 1]
```diff
@@ -1,7 +1,10 @@
 import json
+
+from asgiref.sync import sync_to_async, async_to_sync
 from channels.generic.websocket import AsyncWebsocketConsumer
+from channels.generic.websocket import WebsocketConsumer
```
Comment (bilbosz) [inline game/consumers.py:5]: You can change previous line to `channels.generic.websocket import AsyncWebsocketConsumer, WebsocketConsumer`

File: game/consumers.py [hunk 2]
```diff
@@ -48,3 +51,49 @@ def move_token(self, id, x, y):
 
     async def token_position_changed(self, event):
         await self.send(text_data=json.dumps(event["data"]))
+
+
+class ChatConsumer(WebsocketConsumer):
+    def connect(self):
+        self.room_name = self.scope['url_route']['kwargs']['room_name']
```
Comment (bilbosz) [inline game/consumers.py:58]: Double quotes are preferable to single quotes

File: game/consumers.py [hunk 3]
```diff
@@ -48,3 +51,49 @@ def move_token(self, id, x, y):
...
+        self.room_group_name = 'chat_%s' % self.room_name
```
Comment (bilbosz) [inline game/consumers.py:59]: Let's use "chat_{self.room_name}" instead

File: game/templates/chat/room.html [hunk 4]
```diff
@@ -0,0 +1,42 @@
+<div>
+    <p>Logges as: {{user.username}}</p>
+    <textarea id="chat-log" cols="50" rows="70"></textarea>
+    <input id="chat-message-input" type="text" size="30">
+    <input id="chat-message-submit" type="button" value="Send">
+    <script>
```
Comment (bilbosz) [inline game/templates/chat/room.html:6]: Please move script to static file

File: game/consumers.py (hunk 1 above)
Comment (bilbosz) [inline game/consumers.py:5]: You can change previous line to `channels.generic.websocket import AsyncWebsocketConsumer, WebsocketConsumer` and remove this one

File: treflun/asgi.py [hunk 5]
```diff
@@ -9,8 +9,14 @@
...
 import os
 
+from channels.auth import AuthMiddlewareStack
+from channels.routing import ProtocolTypeRouter, URLRouter
 from django.core.asgi import get_asgi_application
 
 os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'treflun.settings')
 
-application = get_asgi_application()
+# application = get_asgi_application()
```
Comment (bilbosz) [inline treflun/asgi.py:18]: You can remove this line

**Review by bilbosz — CHANGES_REQUESTED** [review]

File: game/consumers.py (hunk 2 above)
Comment (sp2mbn) [reply game/consumers.py]: done

File: game/consumers.py (hunk 3 above)
Comment (sp2mbn) [reply game/consumers.py]: done

File: game/templates/chat/room.html (hunk 4 above)
Comment (sp2mbn) [reply game/templates/chat/room.html]: moved

File: game/consumers.py (hunk 1 above)
Comment (sp2mbn) [reply game/consumers.py]: moved

File: treflun/asgi.py (hunk 5 above)
Comment (sp2mbn) [reply treflun/asgi.py]: removed

**Review by bilbosz — APPROVED** [review]

**Pull request merged**